compiler/rustc_expand/src/base.rs

```diff
@@ -745,9 +745,15 @@ impl SyntaxExtension {
             }
         }
 
-        let builtin_name = sess
+        let (builtin_name, helper_attrs) = sess
             .find_by_name(attrs, sym::rustc_builtin_macro)
-            .map(|a| a.value_str().unwrap_or(name));
+            .map(|attr| {
+                parse_macro_name_and_helper_attrs(sess.diagnostic(), attr, "built-in").map_or_else(
+                    || (Some(name), Vec::new()),
+                    |(name, helper_attrs)| (Some(name), helper_attrs),
```

Comment:
if we don't concatenate the original `helper_attrs` list with the new list, we lose the old ones. I don't know if this something that can happen in practice, but we should probably have a comment here explaining why losing that list is ok.

Reply:
It cannot happen, I'll add an assert.

Reply:
k. r=me with that

Reply:
Ugh, it's actually possible when you have both `#[rustc_builtin_macro]` and `#[proc_macro_derive]`.
However it doesn't make sense in practice, so I've only added a comment.